vega_sim/scenario/common/agents.py

```diff
@@ -1125,6 +1129,90 @@ def step(self, vega_state: VegaState):
                 key_name=self.key_name,
             )
 
+    def _scale_orders(self, buy_shape: List[MMOrder], sell_shape: List[MMOrder]):
+
+        est_midprice = 0.5 * (buy_shape[0].price + sell_shape[0].price)
+
+        instantaneous_liquidity = min(
+            [
+                self._calculate_liquidity(
+                    side=vega_protos.SIDE_BUY,
+                    orders=buy_shape,
+                    midprice=est_midprice,
+                ),
+                self._calculate_liquidity(
+                    side=vega_protos.SIDE_SELL,
+                    orders=sell_shape,
+                    midprice=est_midprice,
+                ),
+            ]
+        )
+
+        # Compute the scaling factor
+        stake_to_ccy_siskas = self.vega.get_network_parameter(
```

Comment:
Would getting this at every step of the agent slow things down? Maybe we can just get it once upfront? Though this is of course safer if it somehow gets changed half-way through a run.

Reply:
I'll add a flag so we can change the behaviour:

Reply:
If you call it something along the lines of "sync_state_every_n_steps" (but maybe more concise 😄 ) you can make it a general function which it calls and we can add other things to in future too (might be able to put it in the base class too)

Reply:
Sounds good, does this make sense to do this outside the agent in the environment so we can pass to every agent?

Reply:
No I think there's some stuff that's useful to specific agents only so can be inside the step. Though we do also want to keep general market data in the VegaState object where possible to pass to every agent, but I think network parameters are specific enough to load locally